File: maxDepth/maxDepth.ts
type TreeNode = {
  value: number;
  left?: TreeNode | null;
  right?: TreeNode | null;
};

function maxDepth(root: TreeNode | null | undefined): number {
  if (root == null) {
    return 0;
  }

  return 1 + Math.max(
    maxDepth(root.left),
    maxDepth(root.right)
  );
}

File: maxDepth/maxDepth.py
from dataclasses import dataclass

@dataclass
class TreeNode:
    value: int
    left: "TreeNode | None" = None
    right: "TreeNode | None" = None



def max_depth(root: TreeNode | None) -> int:
    if root is None:
        return 0

    return 1 + max(
        max_depth(root.left),
        max_depth(root.right)
    )
